src/ServiceControl/Infrastructure/RavenDB/Expiration/ErrorMessageCleaner.cs

```diff
@@ -0,0 +1,112 @@
+        {
+            using (DocumentCacher.SkipSettingDocumentsInDocumentCache())
+            using (database.DisableAllTriggersForCurrentThread())
+            using (var cts = new CancellationTokenSource())
+            {
+                var stopwatch = Stopwatch.StartNew();
+                var documentWithCurrentThresholdTimeReached = false;
+                var items = new List<ICommandData>(deletionBatchSize);
+                var attachments = new List<string>(deletionBatchSize);
+                var docsToExpire = 0;
+                try
+                {
+                    var query = new IndexQuery
+                    {
+                        Start = 0,
+                        PageSize = deletionBatchSize,
+                        Cutoff = SystemTime.UtcNow,
+                        FieldsToFetch = new[]
+                        {
+                            "__document_id",
+                            "Last-Modified",
+                            "MessageId"
+                        },
+                        SortedFields = new[]
+                        {
+                            new SortedField("Last-Modified")
+                            {
+                                Field = "Last-Modified",
+                                Descending = false
+                            }
+                        },
+                    };
+                    var indexName = new ExpiryErrorMessageIndex().IndexName;
+                    database.Query(indexName, query, CancellationTokenSource.CreateLinkedTokenSource(database.WorkContext.CancellationToken, cts.Token).Token,
+                        null,
+                        doc =>
+                        {
+                            if (documentWithCurrentThresholdTimeReached)
+                            {
+                                return;
+                            }
+
+                            var id = doc.Value<string>("__document_id");
+                            if (string.IsNullOrEmpty(id))
+                            {
+                                return;
+                            }
+
+                            if (doc.Value<DateTime>("Last-Modified") >= expiryThreshold)
+                            {
+                                documentWithCurrentThresholdTimeReached = true;
+                                cts.Cancel();
+                                return;
+                            }
+
+                            items.Add(new DeleteCommandData
+                            {
+                                Key = id
+                            });
+
+                            attachments.Add(doc.Value<string>("MessageId"));
+                        });
+                }
+                catch (OperationCanceledException)
+                {
+                    //Ignore
+                }
+
+                logger.Debug("Batching deletion of {0} documents.", items.Count);
+
+                docsToExpire += items.Count;
```

Comment:
docsToExpire will always be 0 before this, won't it? We are not in a loop at this point